src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -68,14 +169,15 @@ public TemporaryTextStorage(TemporaryStorageService service)
                     _service = service;
                 }
 
-                public TemporaryTextStorage(TemporaryStorageService service, string storageName, long size, Encoding encoding)
+                public TemporaryTextStorage(TemporaryStorageService service, string storageName, long offset, long size, Encoding encoding)
                 {
                     _service = service;
                     _encoding = encoding;
-                    _memoryMappedInfo = new MemoryMappedInfo(storageName, size);
+                    _memoryMappedInfo = new MemoryMappedInfo(storageName, offset, size);
                 }
 
                 public string Name => _memoryMappedInfo?.Name;
+                public long Offset => _memoryMappedInfo.Offset;
                 public long Size => _memoryMappedInfo.Size;
```

Comment:
I'm not sure. I was just using the existing pattern (predated my initial work on this file).